### PR TITLE
applications: asset_tracker_v2: simplify A-GPS implementation

### PR DESCRIPTION
**Pull request by tokangas**
Changed implementation to use nRF Cloud A-GPS directly to simplify A-GPS usage and configuration.

Signed-off-by: Tommi Kangas <tommi.kangas@nordicsemi.no>

### CODE REVIEW

File: applications/asset_tracker_v2/src/modules/data_module.c [hunk 1]
```diff
@@ -964,29 +964,19 @@ static void agps_request_handle(struct nrf_modem_gnss_agps_data_frame *incoming_
...
-	 * to sent the request. This is handled internally in the A-GPS library.
-	 *
-	 * If CONFIG_AGPS_SRC_SUPL is enabled a SUPL session is started and A-GPS data is requested
-	 * from a configured SUPL server.
-	 *
-	 * If both CONFIG_NRF_CLOUD_MQTT and CONFIG_AGPS_SRC_SUPL are enabled, SUPL will take
-	 * precedence over nRF Cloud as it is selected as the A-GPS source. By default, nRF Cloud is
-	 * selected as the A-GPS source via the CONFIG_AGPS_SRC_NRF_CLOUD option.
+	/* If CONFIG_NRF_CLOUD_MQTT is enabled, the nRF Cloud MQTT transport library will be used
+	 * to sent the request.
```
Comment (jtguggedal) [inline applications/asset_tracker_v2/src/modules/data_module.c:968]: sent -> send

**Review by jhn-nordic — APPROVED** [review]

File: applications/asset_tracker_v2/src/modules/data_module.c [hunk 2]
```diff
@@ -964,29 +964,19 @@ static void agps_request_handle(struct nrf_modem_gnss_agps_data_frame *incoming_
...
-	 *
-	 * If both CONFIG_NRF_CLOUD_MQTT and CONFIG_AGPS_SRC_SUPL are enabled, SUPL will take
-	 * precedence over nRF Cloud as it is selected as the A-GPS source. By default, nRF Cloud is
-	 * selected as the A-GPS source via the CONFIG_AGPS_SRC_NRF_CLOUD option.
+	/* If CONFIG_NRF_CLOUD_MQTT is enabled, the nRF Cloud MQTT transport library will be used
+	 * to send the request.
 	 */
-#if defined(CONFIG_AGPS) && (defined(CONFIG_NRF_CLOUD_MQTT) || defined(CONFIG_AGPS_SRC_SUPL))
-	err = agps_request_send(*incoming_request);
+#if defined(CONFIG_NRF_CLOUD_AGPS) && defined(CONFIG_NRF_CLOUD_MQTT)
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/src/modules/data_module.c:970]: This effectively disables use of SUPL for A-GPS. This is not what it says in the commit.

File: applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst [hunk 3]
```diff
@@ -386,12 +386,6 @@ You can add the following optional configurations to configure the heap or to pr
 * :kconfig:`CONFIG_PDN_DEFAULTS_OVERRIDE` - Used for manual configuration of the APN. Set the option to ``y`` to override the default PDP context configuration.
 * :kconfig:`CONFIG_PDN_DEFAULT_APN` - Used for manual configuration of the APN. An example is ``apn.example.com``.
 
-The application supports Assisted GPS.
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst:389]: Why is this removed? The application still supports A-GPS.

Comment (tokangas) [reply applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst:389]: The section is called "Optional library configurations" and A-GPS is enabled by default, so I didn't see this as necessary anymore.

Comment (coderbyheart) [reply applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst:389]: But `CONFIG_NRF_CLOUD_AGPS` is still an optional setting, and needs extra set up on nRF Cloud side AFAIK: in order to use nRF Cloud A-GPS must be activated manually).

File: applications/asset_tracker_v2/src/modules/data_module.c (hunk 2 above)
Comment (tokangas) [reply applications/asset_tracker_v2/src/modules/data_module.c:970]: This only removes A-GPS library usage (along with SUPL support) from asset_tracker_v2. It doesn't remove the A-GPS library itself, which is still used by asset_tracker and the AGPS sample.

**Comment by tokangas** [comment]
> Needs changelog entry.

Added.
